newsroom/push.py

```diff
@@ -87,17 +121,136 @@ def publish_item(doc):
     return _id
 
 
-@blueprint.route('/push', methods=['POST'])
-def push():
-    assert_test_signature(flask.request)
-    item = flask.json.loads(flask.request.get_data())
-    assert 'guid' in item, {'guid': 1}
-    assert 'type' in item, {'type': 1}
-    orig = app.data.find_one('wire_search', req=None, _id=item['guid'])
-    item['_id'] = publish_item(item)
-    notify_new_item(item, check_topics=orig is None)
-    app.cache.delete(HOME_ITEMS_CACHE_KEY)
-    return flask.jsonify({})
+def publish_event(event):
+    orig = app.data.find_one('agenda', req=None, _id=event['guid'])
```

Comment:
I guess it should be searching by `event_id` not `_id`

Reply:
Ok, in that case this _id could be an autogenerated id for agenda I guess

Reply:
actually using _id will be probably faster, so I would keep it but you can add a comment